### ISSUE TITLE
Suggestion: Set Nickname Command

### ISSUE THREAD

**Issue by Harucho** [jonas747/yagpdb#368]
On my server there are colour roles that give everyone the ability to express themselves with the colour of their name, including mods. This has unfortunately hidden some privileges my mods should have, such as changing other member's nicknames if found inappropriate. Since listing previous nicknames is already an option, what about setting it? It would still require the "Manage Nickname" permission of course.

**Comment by noahggarrett** [comment]
I agree the nicknames feature would be very useful to get around that for the mean while Harucho I would just make a role that has nicking perms and revoke it if a user misuses their username.

**Comment by svalencia014** [comment]
I'll work on trying to make it happen. Will post updates here.

**Comment by LemmeCry** [comment]
Feel free to PR but this is something we're actively against. Making this change is fairly simple, but it's not been done because it makes abusing YAG easier. Discord makes it clear to only change nicknames for moderation purposes and not to harass people, which is something that people would immediately do if they could use a bot to mass change nicknames via yag.

**Comment by svalencia014** [comment]
I just remembered that this bot is in go so It will probably be a while.




> Feel free to PR but this is something we're actively against. Making this change is fairly simple, but it's not been done because it makes abusing YAG easier. Discord makes it clear to only change nicknames for moderation purposes and not to harass people, which is something people would immediately do if they could use a bot to mass change nicknames via yag.

It would be configurable per server. By default would be Mods/Admin roles only, just like the moderation stuff.

**Comment by jo3-l** [comment]
Abuse does not refer to unrestricted usage of the command by those lacking permissions; rather, it refers to abuse of the bot's functionality itself by server managers.

**Comment by svalencia014** [comment]
A probably not so great solution is to put it behind the premium thing. Not sure anyone would pay for it just to abuse the nicknames.

**Comment by LemmeCry** [comment]
Even if they don't just subscribe for nickname changing, the fact that you can do it with premium still doesn't negate the abuse.
